doc: Modify CLI installation guide for Linux, macOS & Windows

ymgupta opened the pull request:
#### Summary
Improvized the CLI installation guide with instructions for the three platforms Linux, macOS & Windows.

#### Changes
- Changed the documentation structure with separate tabs containing installation steps to be performed in Linux, Windows, and macOS.
- Added steps to install CLI in windows.
- Re-organized the `Configuration` & `Generate configuration file` sections.
  - Reason for this is we will generate a configuration file first and then configure the CLI
- Added note to get the cloud-hosted address for CLI configuration.


#### Checklist
<!-- Make sure that this pull request is complete. -->

- [x] Scope: The referenced issue is addressed, there are no unrelated changes.
- [x] Run Locally: Verified that the docs build using `make server`, posted screenshots, verified external links.
- [ ] New Features Marked: Documentation for new features is marked using the `new-in-version` shortcode, according to the guidelines in [CONTRIBUTING](CONTRIBUTING.md).
- [x] Style Guidelines: Documentation obeys style guidelines in [CONTRIBUTING](CONTRIBUTING.md).
- [x] Commits: Commit messages follow guidelines in [CONTRIBUTING](CONTRIBUTING.md), there are no fixup commits left.

trnkrishna commented:
Update the PR description("Changes" section) with what all changes have been done.

ymgupta commented:
`Govardhan`
I have updated the Changes section
- We have changed the documentation structure and added a few instructions & NOTE for windows
- Re-organized a section.

CC: @trnkrishna
